src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,105 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROW_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeeded();
+        int index = calculateIndex(table.length, hash(key));
+        Node<K, V> newNode = new Node<>(key, value);
+
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (current.key == key || key != null && key.equals(current.key)) {
+                    current.value = value;
+                    return;
+                }
+                if (current.next == null) {
+                    current.next = newNode;
+                    break;
+                }
+                current = current.next;
+            }
+        }
 
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        return (node == null) ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int calculateIndex(int length, int hash) {
+        return (length - 1) & hash;
+    }
```

Comment:
```suggestion
    private int calculateIndex(int hash) {
        return (table.length- 1) & hash;
    }
```

Reply:
i can't change hash and calculateIndex methods to your requirements because i can't pass throught this tests
checkArrayLengthAfterResizingTest,
checkTheHashMapIncrease
when I try to implement your requirements, the tests fail